src/fges_helper.jl

```diff
@@ -0,0 +1,242 @@
+ischild(g, x, y) = !has_edge(g,x,y) && has_edge(g,y,x)
+
+
+"""
+    isdescendent(g, edge::Edge)
+    isdescendent(g, x, y)
+Return `true` if `x`←`y` OR `x`-`y` in the graph `g`.
+"""
+isdescendent(g, edge) = has_edge(g,reverse(edge))
+isdescendent(g, x, y) = has_edge(g,y,x)
+
+"""
+    isoriented(g, edge::Edge)
+    isoriented(g, x, y)
+Test if `x` and `y` are connected by a directed edge in the graph `g`, either x←y OR x→y.
+Can also perform the same test given an `edge`.
+"""
+isoriented(g, edge) = has_edge(g,edge) ⊻ has_edge(g,reverse(edge)) # xor
+isoriented(g, x, y) = has_edge(g,x,y) ⊻ has_edge(g,y,x)
+
+
+"""
+    isclique(g, nodes)
+Return `true` if all vertices in `nodes` are connected to each other in the graph `g`.
+"""
+function isclique(g, nodes)
+
+    for nodePair in allpairs(nodes)
+        if !isadjacent(g, nodePair)
+            return false
+        end
+    end
+
+    return true
+end
+
+
+#Do the nodesRemoved block all semi-directed paths between src and dest?
+"""
+    isblocked(g, src, dest, nodesRemoved)
+Return `true` if there is no semi-directed path between `src` and `dest` in the graph `g`.
+A set of vertices (`nodesRemoved`) can be removed from the graph before searching for a semi-directed path.
+A semi-directed path between `src` and `dest` is a list of edges in `g` where every edge is either undirected or points toward `dest`. 
+    src → x₁ - x₂ → dest ✓
+    src → x₁ ← x₂ - dest ✖
+"""
+function isblocked(g, x, y, nodesRemoved)
+
+    #Keep track of all the nodes visited
+    visited = zeros(Bool, nv(g))
+
+    # mark excluded vertices as visited
+    for vᵢ in nodesRemoved 
+        visited[vᵢ] = true
+    end
+
+    #If src or dest were in nodesRemoved, the path is blocked
+    (visited[x] || visited[y]) && return true
+
+    #if the src and dest are the same, path is itself
+    x == y && return false
+
+    #check if scr or dest have no neighbors
+    if countNeighbors(g, x)==0 || countNeighbors(g, y)==0
+        return true
+    end
+
+    queue = [x]
+    visited[x] = true
+    while !isempty(queue)
+        currentNode = popfirst!(queue) # get new element from queue
+        for vᵢ in descendent(g,currentNode)
+            vᵢ == y && return false
+            if !visited[vᵢ]
+                push!(queue, vᵢ) # push onto queue
+                visited[vᵢ] = true
+            end
+        end
+    end
+    return true
+end
+
+
+
+####################################################################
+# Modify Edges
+####################################################################
+
+"""
+    orientedge!(g, x, y)
+Update the edge `x`-`y` to `x`→`y` in the graph `g`. 
+"""
+orientedge!(g, edge) = rem_edge!(g, reverse(edge))
+orientedge!(g, x, y) = rem_edge!(g, y, x)
+
+
+
+####################################################################
+# Neighborhood functions 
+####################################################################
+
+#Should these be non-allocating iterators? 
```

Comment:
Aren't they?

Reply:
I think they weren't originally, but I changed it and forgot to remove the comment.